scripts/linters/custom_eslint_checks/rules/test-message-style.js

```diff
@@ -0,0 +1,121 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Lint check to ensure that correct style is followed for the
+ * test messages in calling of 'it'.
+ */
+
+'use strict';
+
+module.exports = {
+  meta: {
+    type: 'layout',
+    docs: {
+      description: 'There should not be any unused directive',
+      category: 'Stylistic Issues',
+      recommended: false
+    },
+    fixable: null,
+    schema: [],
+    messages: {
+      useShould: 'Test message should start with \'should\'',
+      singleSpace: 'Do not use multiple consecutive spaces in the test message',
+      noSpaceAtEnd: 'Do not use space at the end of test message',
+      spaceAtEnd: (
+        'Use space at the end of test message in case of ' +
+          'binary operator is used')
+    }
+  },
+
+  create: function(context) {
+    var nodePos = {};
+
+    var checkMessageStartsWithShould = function(testMessageNode) {
+      if (!testMessageNode.value.startsWith('should ')) {
+        context.report({
+          testMessageNode,
+          loc: testMessageNode.loc,
+          messageId: 'useShould'
+        });
+      }
+    };
+
+    var checkSpacesInMessage = function(testMessageNode) {
+      if (testMessageNode.value.includes('  ')) {
+        context.report({
+          testMessageNode,
+          loc: testMessageNode.loc,
+          messageId: 'singleSpace'
+        });
+      }
+    };
+
+    var checkNoSpaceAtEndOfMessage = function(testMessageNode) {
+      if (testMessageNode.value.endsWith(' ')) {
+        context.report({
+          testMessageNode,
+          loc: testMessageNode.loc,
+          messageId: 'noSpaceAtEnd'
+        });
+      }
+    };
+
+    var checkSpaceAtEnd = function(testMessageNode) {
+      if (!testMessageNode.value.endsWith(' ')) {
+        context.report({
+          testMessageNode,
+          loc: testMessageNode.loc,
+          messageId: 'spaceAtEnd'
+        });
+      }
+    };
+
+    var checkMessage = function(node) {
```

Comment:
Can we try to have this pattern to avoid duplicate checks and readable code? (If it takes time to implement this then please leave a code comment explaining the current structure!) 

```
checkMessageInBinExp (node):
   checkMessageNode(left)
   checkMessageNode(rigt)

checkMessageInLinteral(node):
   ....
   ....


checkMessageNode(node)
if node is liter:
 checkMessageInLinteral
if node is BEx:
 checkMessageInBinExp
```

Reply:
Done.